### DISCUSSION TITLE
Pagination not refreshed after component render based on filter

### DISCUSSION THREAD

**Discussion by Vladjen** [livewire/livewire#3623] [1. Help]
I am filtering a model with the help of checkboxes. The query is paginated and the pagination works all right. The filter works fine as well. 

Problem : When in the first results page, the filter works fine and the results are shown (`15 paginated results are shown based on my model query`). When not in the first result page (2,3 etc) and filtered, the results are not shown and stays on the same page resulting blank  component (`if only 3 results match my filter then, it shows on the first page while i am not in the first page`). However when i go back to the first page, the results are shown.  The pagination does not refresh, stays on the same url.

```php

$this->teachers = User::where('user_type','tutor')
              ->orderBy('created_at','desc')
              ->whereHas('Teacher', function($query) {
                        $query->when($this->payment_term, function($query) {
                                return  $query->whereIn('fee->payment_term',$this->payment_term);
                        });
              })->paginate(15);
```

i tried refreshing but no use of it. Wonder what i am doing wrong here. Any help would be greatly appreciated.

**Comment by joshhanley** [comment]
@Vladjen hmm I think if you filter your data, you should manually reset your paginator page, so that way it's always viewing the first page after the filters are applied.

I think you can do that by calling `$this->resetPage()`.

Hope this helps!

**Comment by scottzirkel** [comment]
`$this->resetPage()` works fine, unless you are specifying the page name as outlined in the docs under [Multiple Paginators](https://laravel-livewire.com/docs/2.x/pagination#multiple-paginators). Then you should pass through your custom page name.
